server/jetstream_benchmark_test.go

```diff
@@ -1174,3 +1174,184 @@ func BenchmarkJetStreamKV(b *testing.B) {
 		)
 	}
 }
+
+func BenchmarkJetStreamObjStore(b *testing.B) {
+	const (
+		verbose            = false
+		objStoreNamePrefix = "B_"
+		keyPrefix          = "K_"
+		seed               = 12345
+		initKeys           = true
+
+		// read/write ratios
+		ReadOnly  = 1.0
+		WriteOnly = 0.0
+	)
+
+	// benchmark options matrix
+	benchmarkOptions := struct {
+		replicaAndClusterSize []int
+		numKeys               []int
+		objectSizeRange       [][]int
+		rwRatio               []float64
+		storageType           []nats.StorageType
+	}{
+		[]int{1, 3},            // 1 replica, 3 replicas
+		[]int{10, 1000, 10000}, // 10 keys, 1K keys, 10K keys
+		[][]int{
+			{1024, 1024},     // 1KB byte objects
+			{102400, 102400}, // 1MB byte objects
+			{1024, 102400},   // 1KB - 1MB byte objects
+		},
+		[]float64{ReadOnly, WriteOnly, 0.8},                      // readOnly, writeOnly, 80% read
+		[]nats.StorageType{nats.MemoryStorage, nats.FileStorage}, // memory retention, file retention
+	}
```

Comment:
This is pretty confusing. I think you should have tables with values, and split into separate tables only if you want to dot-product. Like the rest of the benchmarks do.

Reply:
I might be wrong but I thought we had talked about a matrix for the benchmark cases? No problem, I can change it to the tabled method that the other benchmarks have.

 I'm a little unsure what the format for separate tables that will then be taken the dot product of. Would it look like this?
 ```
inMem = {
  {opt1, opt2}
  {[min: 1, max: 2], [min:10, max:20], ...} 
}
// run inMem benchmark cases matrix, (in memory storage is constant)

fileStore = {
  {opt3, opt4}
  {[min: 1000, max: 2200], [min:310, max:420], ...} 
}
// run fileStore benchmark cases matrix (filestore storage is constant)
```